src/librustc_trans/trans/_match.rs

```diff
@@ -1624,21 +1624,9 @@ pub fn store_arg<'blk, 'tcx>(mut bcx: Block<'blk, 'tcx>,
         Some(ident) => {
             // Generate nicer LLVM for the common case of fn a pattern
             // like `x: T`
-            let arg_ty = node_id_type(bcx, pat.id);
-            if type_of::arg_is_indirect(bcx.ccx(), arg_ty)
-                && bcx.sess().opts.debuginfo != FullDebugInfo {
-                // Don't copy an indirect argument to an alloca, the caller
-                // already put it in a temporary alloca and gave it up, unless
-                // we emit extra-debug-info, which requires local allocas :(.
```

Comment:
So is this PR a trade-off, or did LLVM get better than it was back when this comment was written?
Also, if we always copy, that means we don't need to create new copies in the callers - which should always be an improvement if the callee copies aren't mutated.

IMO, we should never copy a value that moves, even if there are mutations, and I'm not sure we can teach LLVM about what is effectively `&move T`.